### PR TITLE
feat: add a new thread pool to process slow query

### PR DESCRIPTION
**Pull request by Smityz**
Related PR: https://github.com/apache/incubator-pegasus/pull/782

### CODE REVIEW

**Review by levy5307 — APPROVED** [review]

File: include/dsn/dist/replication/replication.codes.h [hunk 1]
```diff
@@ -37,12 +37,15 @@ DEFINE_THREAD_POOL_CODE(THREAD_POOL_COMPACT)
 DEFINE_THREAD_POOL_CODE(THREAD_POOL_INGESTION)
 DEFINE_THREAD_POOL_CODE(THREAD_POOL_SLOG)
 DEFINE_THREAD_POOL_CODE(THREAD_POOL_PLOG)
+DEFINE_THREAD_POOL_CODE(THREAD_SLOW_QUERY)
```
Comment (Shuo-Jia) [inline include/dsn/dist/replication/replication.codes.h:40]: why name `THREAD_SLOW_QUERY`

Comment (Shuo-Jia) [reply include/dsn/dist/replication/replication.codes.h:40]: And the name should like `THREAD_POOL_...`

File: include/dsn/dist/replication/replication.codes.h [hunk 2]
```diff
@@ -37,12 +37,15 @@ DEFINE_THREAD_POOL_CODE(THREAD_POOL_COMPACT)
...
 
 #define DEFINE_STORAGE_WRITE_RPC_CODE(x, allow_batch, is_idempotent)                               \
     DEFINE_STORAGE_RPC_CODE(                                                                       \
         x, TASK_PRIORITY_LOW, THREAD_POOL_REPLICATION, true, allow_batch, is_idempotent)
 #define DEFINE_STORAGE_READ_RPC_CODE(x)                                                            \
     DEFINE_STORAGE_RPC_CODE(x, TASK_PRIORITY_COMMON, THREAD_POOL_LOCAL_APP, false, true, true)
+#define DEFINE_SCAN_STORAGE_READ_RPC_CODE(x)                                                       \
```
Comment (Shuo-Jia) [inline include/dsn/dist/replication/replication.codes.h:47]: `DEFINE_STORAGE_SCAN_RPC_CODE`

File: include/dsn/dist/replication/replication.codes.h (hunk 1 above)
Comment (Smityz) [reply include/dsn/dist/replication/replication.codes.h:40]: THREAD_POOL_SCAN ?

**Review by Shuo-Jia — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]
Lgtm I've leave some comments on Pegasus repo. Anyway This pr is a relatively short sighted improvement to me.

**Review by Shuo-Jia — APPROVED** [review]

File: include/dsn/dist/replication/replication.codes.h [hunk 3]
```diff
@@ -37,12 +37,15 @@ DEFINE_THREAD_POOL_CODE(THREAD_POOL_COMPACT)
...
 DEFINE_THREAD_POOL_CODE(THREAD_POOL_PLOG)
+DEFINE_THREAD_POOL_CODE(THREAD_POOL_SCAN)
 
 #define DEFINE_STORAGE_WRITE_RPC_CODE(x, allow_batch, is_idempotent)                               \
     DEFINE_STORAGE_RPC_CODE(                                                                       \
         x, TASK_PRIORITY_LOW, THREAD_POOL_REPLICATION, true, allow_batch, is_idempotent)
 #define DEFINE_STORAGE_READ_RPC_CODE(x)                                                            \
     DEFINE_STORAGE_RPC_CODE(x, TASK_PRIORITY_COMMON, THREAD_POOL_LOCAL_APP, false, true, true)
+#define DEFINE_STORAGE_SCAN_RPC_CODE(x)                                                            \
+    DEFINE_STORAGE_RPC_CODE(x, TASK_PRIORITY_COMMON, THREAD_SLOW_QUERY, false, true, true)
```
Comment (hycdong) [inline include/dsn/dist/replication/replication.codes.h:48]: `THREAD_SLOW_QUERY`? Maybe `THREAD_POOL_SCAN`

**Review by hycdong — CHANGES_REQUESTED** [review]

File: include/dsn/dist/replication/replication.codes.h (hunk 3 above)
Comment (Smityz) [reply include/dsn/dist/replication/replication.codes.h]: fixed

**Review by hycdong — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]

**Pull request merged**